recipes/cpr/all/conanfile.py

```diff
@@ -86,7 +86,8 @@ def requirements(self):
         # FIXME: This is a very dirty hack.
         # with_ssl == _AUTO_SSL, cpr needs the openssl header to compile. But Conan recipe does not know which SSL library to use.
         # because cpr's CMakeLists.txt automatically detects SSL libraries with CPR_ENABLE_SSL == ON.
-        if self.settings.os in ["Linux", "FreeBSD"] and self.options.with_ssl in ["openssl", CprConan._AUTO_SSL]:
+        if ((self.settings.os in ["Linux", "FreeBSD", "Android"] and self.options.with_ssl == CprConan._AUTO_SSL) or
```

Comment:
In another PR, I would suggest to remove this ugly `_AUTO_SSL` and `_OFF` (leading to convoluted and fragile logic, this recipe can still fail with other OS not listed here), and keep things under control.

In upstream CMakeLists, when autodetection of SSL is enabled, it leads to:

- Windows => winssl
- Apple OS => darwinssl
- otherwise => OpenSSL (or MbedTLS)

Something under control should look like this:

```python
    options = {
        (...)
        "with_ssl": [False, "openssl", "darwinssl", "winssl"],
        (...)
    }
    default_options = {
        (...)
        "with_ssl":  "openssl",
        (...)
    }

    (...)

    def config_options(self):
        (...)
        if self.settings.os == "Windows":
            self.options.with_ssl = "winssl"
        elif is_apple_os(self):
            self.options.with_ssl = "darwinssl"

    (...)

    def requirements(self):
        (...)
        if self.options.with_ssl == "openssl":
            self.requires("openssl/[>=1.1 <4]")

    def validate(self):
        (...)
        if ssl_library in ("openssl", "darwinssl") and ssl_library != self.dependencies["libcurl"].options.with_ssl:
            raise ConanInvalidConfiguration(
                f"{self.ref}:with_ssl={self.options.with_ssl} requires libcurl:with_ssl={self.options.with_ssl}"
            )
        (...)

    (...)

    def generate(self):
       (...)
        tc.variables["CPR_FORCE_DARWINSSL_BACKEND"] = self.options.with_ssl == "darwinssl"
        tc.variables["CPR_FORCE_OPENSSL_BACKEND"] = self.options.with_ssl == "openssl"
        tc.variables["CPR_FORCE_WINSSL_BACKEND"] = self.options.with_ssl == "winssl"
        tc.variables["CMAKE_USE_OPENSSL"] = (self.options.with_ssl == "openssl")
        tc.variables["CPR_FORCE_MBEDTLS_BACKEND"] = False # TODO: add mbedtls backend
        tc.variables["CPR_ENABLE_SSL"] = bool(self.options.with_ssl)
```

Reply:
I have updated the code as you suggested.

Something I still find surprising with this recipe on Windows is that libcurl uses OpenSSL as default, and CPR uses winssl. You either have to set libcurl/*:with_ssl or cpr/*:with_ssl for windows to build. Before the refactor the check that libcurl and cpr use the same SSL did not trigger due to AUTOSSL weirdness.

If I change CPR default to OpenSSL to match libcurl that could break build for projects that depended on default being winssl.